sunpy/map/tests/test_mapbase.py

```diff
@@ -821,15 +821,78 @@ def test_quicklook(aia171_test_map):
         assert aia171_test_map._repr_html_() in html_string
 
 
-def test_submap_non_kwarg_parameters(aia171_test_map):
-    bl = SkyCoord(-300*u.arcsec, -300*u.arcsec, frame=aia171_test_map.coordinate_frame)
-    tr = SkyCoord(500*u.arcsec, 500*u.arcsec, frame=aia171_test_map.coordinate_frame)
+def test_submap_inputs(generic_map):
+    bl_pix = [3, 2] * u.pix
+    bl_coord = SkyCoord(20, -10, unit=u.arcsec,
+                        frame=generic_map.coordinate_frame)
+
+    tr_pix = [5, 4] * u.pix
+    tr_coord = SkyCoord(0, 10, unit=u.arcsec,
+                        frame=generic_map.coordinate_frame)
+
+    bl_tr_coord = SkyCoord([20, 0], [-10, 10], unit=u.arcsec,
+                           frame=generic_map.coordinate_frame)
+
+    width_pix = 2 * u.pix
+    height_pix = 2 * u.pix
+    width_deg = 20 * u.arcsec
+    height_deg = 20 * u.arcsec
+
+    # works
+    smap = generic_map.submap(bl_coord, top_right=tr_coord)
+    # assert u.allclose(smap.dimensions, (1, 1) * u.pix)
```

Comment:
These asserts don't work because of the issues being worked on in https://github.com/sunpy/sunpy/pull/4134 but they all should give a (2, 2) map with the same reference pix etc.

Reply:
It does work with (2, 2). For the time being, do we change it to (2, 2)?